internal/gen/utils/gen_python.go

```diff
@@ -0,0 +1,145 @@
+package utils
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+	"text/template"
+)
+
+// {{ range .Imports}}
+// 	import {{ if .Wildcard }}* as {{ .WildcardName }}{{ else }}{{{ range .Imports }}{{.}}{{end}}}{{end}} from '{{ .PackageName }}'
+// 	{{ end }})
+
+var pythonFileTemplate = `# Code generated by the gen package. DO NOT EDIT.
+# *** WARNING: Do not edit by hand unless you're certain you know what you are doing! ***
+from __future__ import annotations
+from typing import Union, Literal, List, Dict, Any, Optional, TypedDict, NotRequired
+from pydantic import BaseModel, Field
+
+{{.Contents.String}}
+`
+
+type PythonFile struct {
+	Name     string
+	Contents *codeBlock
+}
+
+func (p PythonFile) Write() error {
+	tmp := template.Must(template.New("pyfile").Parse(pythonFileTemplate))
+
+	file, err := os.Create(p.Name)
+	if err != nil {
+		return fmt.Errorf("creating file: %v", err)
+	}
+	defer file.Close()
+
+	err = tmp.Execute(file, p)
+	if err != nil {
+		return fmt.Errorf("writing file: %v", err)
+	}
+
+	return nil
+}
+
+func NewPythonFile(
+	name string,
+	contents *codeBlock,
+) PythonFile {
+	return PythonFile{
+		Name:     name,
+		Contents: contents,
+	}
+}
+
+var pythonClassTemplate = `{{if ne "" .Description}}{{printf "# %s\n" .Description}}{{end}}class {{.Name}}(BaseModel):{{ range .Items}}
+    {{if ne "" .Description}}{{printf "# %s\n    " .Description}}{{end}}{{.Name}}: {{if eq false .Required }}Optional[{{.Value}}]{{if .Alias}} = Field(serialization_alias='{{.Alias}}'{{if eq false .Required }}, default=None{{end}}){{else}} = None{{end}}{{else}}{{.Value}}{{if .Alias}} = Field(serialization_alias='{{.Alias}}'{{if eq false .Required }}, default=None{{end}}){{end}}{{end}}{{end}}
+
+    @classmethod
+    def from_dict(cls, data: {{.Name}}Dict) -> {{.Name}}:
+        pipeline_if = {'pipeline_if': data['if']} if 'if' in data else {}
```

Comment:
I'd really love to be able to ship this with ~0 breaking changes if we could. 🙏🏻 Can we keep `step_if`? (Apologies as I don't have the fully context on the "why" on this one, but it seems like it should be reasonable.)